Source/Core/Runtime/PSharpRuntime.cs

```diff
@@ -473,26 +469,25 @@ internal virtual bool CheckStartEventHandler(Machine machine)
 
         #endregion
 
-        #region Timers
+        #region timers
 
         /// <summary>
-        /// Overrides the machine type for instantiating timers
+        /// Creates a new timer that sends a <see cref="TimerElapsedEvent"/> to its owner machine after the specified
+        /// due time. The timer can be configured to periodically timeout, by specifying a non-negative period of time.
+        /// The timer accepts an optional payload to be used during timeout.
         /// </summary>
-        /// <param name="timerMachineType">Machine type</param>
-        public void SetTimerMachineType(Type timerMachineType)
-        {
-            this.Assert(timerMachineType.IsSubclassOf(typeof(Machine)), "TimerMachine must be a subclass of Machine");
-            this.TimerMachineType = timerMachineType;
-        }
+        /// <param name="owner">The owner machine.</param>
+        /// <param name="dueTime">The amount of time to wait before sending the first timeout event.</param>
+        /// <param name="period">The time interval between timeout events. A negative value disables periodic timeout events.</param>
+        /// <param name="name">Optional unique name of the timer.</param>
+        /// <param name="payload">Optional payload of the timeout event.</param>
+        /// <returns>The machine timer.</returns>
+        internal abstract IMachineTimer CreateMachineTimer(Machine owner, TimeSpan dueTime, TimeSpan period, string name, object payload);
 
         /// <summary>
-        /// Return the timer machine type
+        /// Returns the timer machine type.
         /// </summary>
-        /// <returns></returns>
-        internal virtual Type GetTimerMachineType()
-        {
-            return TimerMachineType;
-        }
+        internal abstract Type GetTimerMachineType();
```

Comment:
This method should be marked deprecated? I don't think its used anymore.

Reply:
I only added deprecated to public APIs.